samples/tfm/tfm_secure_peripheral/boards/nrf5340dk_nrf5340_cpuapp.conf

```diff
@@ -0,0 +1,3 @@
+
+CONFIG_NRF_GPIO0_PIN_MASK_SECURE=0x00800000 # P0: 23
+CONFIG_NRF_GPIO1_PIN_MASK_SECURE=0x0000a000 # P1: 13, 15
```

Comment:
@mbolivar-nordic Let me know if you can come up with something better than this.

Reply:
> @mbolivar-nordic Let me know if you can come up with something better than this.

Can you put these into devicetree as properties?

Reply:
I don't know how to do that in a way that is accesible in Kconfig.

I need to provide this to a different build system so our options are limited by that, simplest is to provide everything in a simple define.

Reply:
> I don't know how to do that in a way that is accesible in Kconfig.

What do you mean? That's what many of the helpers in kconfigfunctions.py are for.

Note also that direct access to the devicetree in the build system is available in https://github.com/nrfconnect/sdk-zephyr/blob/bb39dfe3121a1c245d7c5804de0e14e19b299b83/cmake/modules/extensions.cmake#L2490

Reply:
Mostly I mean that I'm not that skilled in DT to know how to do that.

This one seems a bit more complicated than most uses though, because I want a bitmask of all of the GPIOs, so I would need to aggregate many symbols then, wouldn't I?

Also I'm unsure how good idea this is since DPPI channels work the same way, and I don't recall seeing DPPI channels being used in DT, it would be best to have them be configured in the same way I think.

Reply:
> Mostly I mean that I'm not that skilled in DT to know how to do that.

I don't know the problem area so I may be missing something, but here is what I recommend:

https://github.com/mbolivar-nordic/sdk-zephyr/commit/ac6983bfdf2681cf7001fb9eb209cf051ce97a48

Then you can get the value out into Kconfig if you really need it there with kconfigfunctions. But I am not sure why you need it in Kconfig. It looks like you could do something like this to access the property directly instead:

```c
 #ifdef NRF_P0
-		PERIPH("NRF_P0", NRF_P0, CONFIG_SPM_NRF_P0_NS),
+		PERIPH("NRF_P0", NRF_P0, DT_PROP_OR(DT_NODELABEL(gpio0), secure_pin_mask, 0)),
 #endif
```

And similarly for port 1, just use `DT_NODELABEL(gpio1)` instead.

> This one seems a bit more complicated than most uses though, because I want a bitmask of all of the GPIOs, so I would need to aggregate many symbols then, wouldn't I?

> Also I'm unsure how good idea this is since DPPI channels work the same way, and I don't recall seeing DPPI channels being used in DT, it would be best to have them be configured in the same way I think.

I think that's likely just a historical accident; if we were configuring these today they'd probably belong in DT, I guess, since that's where boot-time hardware configuration is supposed to go.

Reply:
Offline discussion: We will use Kconfig for now, switch to DT later when it supports the use-case better.